lib/library.rb

```diff
@@ -0,0 +1,49 @@
+require './lib/author'
+
+class Library
+  attr_reader :name, :authors
+
+  def initialize(name)
+    @name = name
+    @authors = []
+  end
+
+  def add_author(author)
+    @authors << author
+  end
+
+  def books
+    if @authors != []
+      @authors.map do |author|
+        author.books
+      end.flatten
+    else
+      []
+    end
+  end
+
+  def find_author(search_author)
+    @authors.find do |author|
+      author == search_author
+    end
+  end
+
+  def last_book(search_author)
+    find_author(search_author).books.max_by do |book|
+      book.publication_year.to_i
+    end
+  end
+
+  def first_book(search_author)
+    find_author(search_author).books.min_by do |book|
```

Comment:
Going off my last comment, here you could just do search_author.books.min_by....